lightning-invoice/src/payment.rs

```diff
@@ -150,17 +150,16 @@ use std::time::{Duration, SystemTime};
 /// See [module-level documentation] for details.
 ///
 /// [module-level documentation]: crate::payment
-pub struct InvoicePayer<P: Deref, R, S: Deref, L: Deref, E>
+pub struct InvoicePayer<P: Deref, S: Score, R: Deref, SR: Deref<Target = MultiThreadedLockableScore<S>>, L: Deref, E: Deref>
 where
 	P::Target: Payer,
-	R: for <'a> Router<<<S as Deref>::Target as LockableScore<'a>>::Locked>,
-	S::Target: for <'a> LockableScore<'a>,
+	R::Target: Router<S>,
 	L::Target: Logger,
-	E: EventHandler,
+	E::Target: EventHandler,
```

Comment:
Is this also needed for use in bindings?

Reply:
Ah, no, will add a commit that fixes this another way.